### PR TITLE
Unify settings names 

### PR DESCRIPTION
**Pull request by Asnelchristian**

### CODE REVIEW

**Comment by Adrianzatreanu** [comment]
ack dc5bb7ebbe24

**Comment by Asnelchristian** [comment]
reack https://github.com/coala-analyzer/coala-bears/pull/683/commits/cee9e18191c1584fc47b6aa1c436e0a883bfbc1c

**Comment by Adrianzatreanu** [comment]
ack becdf764cc3574135ff81baa659885910d07e5d0

**Comment by Adrianzatreanu** [comment]
ack 8b2a21d295bf558c50db499f1e1c1e9adac20ced

**Comment by Adrianzatreanu** [comment]
ack 3946808c37988485557eaa6f62d6b239bf2f0fdf

File: bears/js/JSHintBear.py [hunk 1]
```diff
@@ -138,9 +138,8 @@ def generate_config(filename, file,
             This option allows the use of the comma operator.
         :param allow_non_breaking_whitespace:
             Allows "non-breaking whitespace characters".
-        :param prohibit_new:
-            This option prohibits the use of constructor functions for
-            side-effects.
+        :param allow_constructor_functions:
+            Allows the use of constructor functions..
```
Comment (Adrianzatreanu) [inline bears/js/JSHintBear.py]: 2 periods

File: bears/js/JSHintBear.py [hunk 2]
```diff
@@ -140,8 +140,8 @@ def generate_config(filename, file,
             Allows "non-breaking whitespace characters".
         :param allow_constructor_functions:
             Allows the use of constructor functions..
-        :param prohibit_undefined:
-            This option prohibits the use of explicitly undeclared variables.
+        :param allow_undefined_variables:
+            Allows the use of explicitly undeclared variables.
```
Comment (Adrianzatreanu) [inline bears/js/JSHintBear.py]: how can you allow the use of explicitly UNDECLARED variabiles? im so confused about this property

**Comment by Adrianzatreanu** [comment]
ack 66884d4780fa143ac8095d01a3a980a4be92b18e

File: bears/js/JSHintBear.py [hunk 3]
```diff
@@ -145,8 +145,8 @@ def generate_config(filename, file,
         :param allow_grouping_operator:
             This option allows the use of the grouping operator when it is
             not strictly required.
-        :param prohibit_variable_statements:
-            This option forbids the use of VariableStatements.
+        :param allow_variable_statements:
+            Allows the use of VariableStatements.
```
Comment (Adrianzatreanu) [inline bears/js/JSHintBear.py]: the use of what? is the case intentionally :D ?

**Comment by Adrianzatreanu** [comment]
ack 6d6952ce716859a52462a0ac65510331dcd0edc0

**Comment by Adrianzatreanu** [comment]
ack bb4aac11f05296710906581b78a8d25b3f71fd3f

**Comment by Adrianzatreanu** [comment]
ack 21d78d41ed41404fd08652ca703191587bff8972

**Comment by Adrianzatreanu** [comment]
ack 3ad21d5ca7acba4d0d72bb1458c33c9c098c0b83

**Comment by Adrianzatreanu** [comment]
ack 4d67607e3591addf0a812e39e4bc817d3750f868

**Comment by Asnelchristian** [comment]
reack https://github.com/coala-analyzer/coala-bears/pull/683/commits/90a8fe43453cea6c122a2884fd00f4230f4ecb50 https://github.com/coala-analyzer/coala-bears/pull/683/commits/0513d214242d1abe8d041ada259cb26180afdb79  https://github.com/coala-analyzer/coala-bears/pull/683/commits/75eeb830c594e5aa9f263d6c680d7e68b24039a4 https://github.com/coala-analyzer/coala-bears/pull/683/commits/f15aff0708d31bb5182941f39f0203121842213a  https://github.com/coala-analyzer/coala-bears/pull/683/commits/dde5306c8365bd180ad4247f7a162e5613a06346 https://github.com/coala-analyzer/coala-bears/pull/683/commits/8fec15ff942e2a2b8a83fa23577b69a192bb0ea5

**Comment by Adrianzatreanu** [comment]
ack fd1655541841c7929317e7c0795c301b2b616446

File: bears/js/JSHintBear.py [hunk 4]
```diff
@@ -54,7 +54,6 @@ def generate_config(filename, file,
                         allow_comma_operator: bool=True,
```
Comment (Adrianzatreanu) [inline bears/js/JSHintBear.py]: this option** in commit message

File: bears/js/JSHintBear.py [hunk 5]
```diff
@@ -142,8 +142,9 @@ def generate_config(filename, file,
         :param allow_grouping_operator:
             This option allows the use of the grouping operator when it is
             not strictly required.
-        :param prohibit_variable_statements:
-            This option forbids the use of VariableStatements.
+        :param allow_var_statement:
+            Allows the use of the ``var`` statement while declaring variable.
```
Comment (Adrianzatreanu) [inline bears/js/JSHintBear.py]: grammar mistake. it should be ``while declaring a variable`` i think

File: bears/js/JSHintBear.py [hunk 6]
```diff
@@ -142,8 +142,9 @@ def generate_config(filename, file,
...
+            Use ``let`` or ``const`` otherwise.
```
Comment (Adrianzatreanu) [inline bears/js/JSHintBear.py]: what do you mean otherwise? its a little unclear. otherwise as in if its False and not true?

**Comment by Asnelchristian** [comment]
reack https://github.com/coala-analyzer/coala-bears/pull/683/commits/d1e386b0c6b6f65889859f389081112657e9fdaa

**Comment by Adrianzatreanu** [comment]
ack c4fa9336f52d3613a4cd05d263dde5ffbf4ebc5d

File: bears/js/JSHintBear.py [hunk 7]
```diff
@@ -142,8 +142,9 @@ def generate_config(filename, file,
         :param allow_grouping_operator:
             This option allows the use of the grouping operator when it is
             not strictly required.
-        :param prohibit_variable_statements:
-            This option forbids the use of VariableStatements.
+        :param allow_var_statement:
+            Allows the use of the ``var`` statement while a declaring variable.
```
Comment (Adrianzatreanu) [inline bears/js/JSHintBear.py]: while a declaring variable :D ?

File: bears/js/JSHintBear.py [hunk 8]
```diff
@@ -142,8 +142,9 @@ def generate_config(filename, file,
...
+            Use ``let`` or ``const`` when it is ``False``.
```
Comment (Adrianzatreanu) [inline bears/js/JSHintBear.py]: better: "Should use let or const while it is False."

File: bears/js/JSHintBear.py [hunk 9]
```diff
@@ -142,8 +142,9 @@ def generate_config(filename, file,
         :param allow_grouping_operator:
             This option allows the use of the grouping operator when it is
             not strictly required.
-        :param prohibit_variable_statements:
-            This option forbids the use of VariableStatements.
+        :param allow_var_statement:
+            Allows the use of the ``var`` statement while declaring a variable.
+            Use ``let`` or ``const`` when it is ``False``.
```
Comment (Adrianzatreanu) [inline bears/js/JSHintBear.py]: better: ``Should uses let or const while it is set to False``. ?

**Comment by Adrianzatreanu** [comment]
ack 3e80c0651f8b64828d3558fcd34403160099a3f7

**Comment by sils1297** [comment]
hey @Adrianzatreanu @Asnelchristian  can you please recheck those commits? The name change sometimes inverts the meaning and then the doc comment doesn't make sense anymore - which is a major issue!

**Comment by Asnelchristian** [comment]
@rultor merge

**Pull request merged**